### PR TITLE
feat: Add simple `nv24` migration

### LINKED ISSUE #295: Skeleton for nv24 to support development and testing
Write the skeleton for nv24, now that the development, testing, and deployment of network version 23 (nv23) nears completion, with the Filecoin Mainnet scheduled to upgrade to nv23 on August 6, 2024, at 12:00 UTC.

This is part of a series of tickets across multiple repos, as building up a network upgrade skeleton requires changes across:
- https://github.com/filecoin-project/ref-fvm/issues/2028
- https://github.com/filecoin-project/filecoin-ffi/issues/478
- Go-State-Types (this ticket)
- https://github.com/filecoin-project/lotus/issues/12351

How to create a network skeleton in Go-State-Types is documented here: https://github.com/filecoin-project/lotus/blob/master/documentation/misc/Building_a_network_skeleton.md#go-state-types-checklist

## Done Criteria
- There is GST developer release that supports the skeleton for network version 24.

## User/Customer
Implementers

### PR DESCRIPTION
**Pull request by kamuik16**
Closes #295
ref https://github.com/filecoin-project/lotus/blob/master/documentation/misc/Building_a_network_skeleton.md#go-state-types-checklist and https://github.com/filecoin-project/go-state-types/pull/258

### CODE REVIEW

**Comment by kamuik16** [comment]
cc @rjan90

File: builtin/v15/migration/top.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
+package migration
+
+import (
+	"context"
+
+	adt15 "github.com/filecoin-project/go-state-types/builtin/v15/util/adt"
+
+	system14 "github.com/filecoin-project/go-state-types/builtin/v14/system"
```
Comment (rjan90) [inline builtin/v15/migration/top.go:8]:
```suggestion
	system15 "github.com/filecoin-project/go-state-types/builtin/v15/system"
```

**Review by rjan90 — COMMENTED** [review]
One small nit, else it looks fine

File: builtin/v15/migration/top.go (hunk 1 above)
Comment (kamuik16) [reply builtin/v15/migration/top.go:8]: I took a look at https://github.com/filecoin-project/go-state-types/pull/258 and it was using a system version 13 for v14 so I assumed it should use one lesser version of system.

Is my assumption correct here or am I missing something?

Comment (rjan90) [reply builtin/v15/migration/top.go:8]: Oh, sorry - you are totally correct! I should have gotten a bit more coffee in the morning before starting reviewing 😅

**Review by rjan90 — APPROVED** [review]

**Comment by rjan90** [comment]
I will ship a v0.15.0-dev release once the checks passes and this has been merged to master, so that you can easily bubble it up when you start working on: https://github.com/filecoin-project/lotus/issues/12351

**Pull request merged**